mongo/document/base.go

```diff
@@ -13,11 +13,11 @@ type Base struct {
 	UpdatedAt time.Time     `bson:"updated_at" json:"updated_at"`
 }
 
-func (d Base) IsPersisted() bool {
+func (d *Base) IsPersisted() bool {
 	return !d.CreatedAt.IsZero()
 }
 
-func (d Base) getID() bson.ObjectId {
```

Comment:
No need of pointer, nothing is modified

Reply:
Then document `Base` does not satisfy the `document` interface as `getID` is implemented for `Base` but the `ensure*` methods are for pointer of Base.

Reply:
Go is smarter than that :P If the struct is passed as a pointer it can access non-pointer methods: https://play.golang.com/p/NicC6AlMbd7

(Not the over way around though: https://play.golang.com/p/o-eDYPih2V4)

Reply:
Yep but interface does not work that way. If you have the following struct and interface:

```go
type T struct {}

type I interface {
  A()
  B()
}
```

And you want the struct `T` to implement the interface `I`, you can NOT do the following:

```go
func (t T) A() {}
func (t *T) B() {}
```

With this example, the struct `T` does not satisfy the interface `I`